app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -126,25 +130,65 @@ if [ x$ID = xrhel ] || [ x$ID = xcentos ] || [ x$ID = xol ]; then
       yum install -y --setopt=obsoletes=0 subscription-manager
     fi
 
-    CONSUMER_RPM=$(mktemp --suffix .rpm)
-    curl --silent --show-error --output $CONSUMER_RPM <%= subscription_manager_configuration_url(hostname: @url_host) %>
+    # Prepare SSL certificate
+    cp -f $SSL_CA_CERT $KATELLO_SERVER_CA_CERT
+    chmod 644 $KATELLO_SERVER_CA_CERT
+
+    # Prepare subscription-manager
+    yum remove -y katello-ca-consumer*
 
-    # Workaround for systems with enabled FIPS,
-    # where installation of RPM generated on RHEL7 cause 'no digest' error
-    # See https://projects.theforeman.org/issues/32068
-    if [ "$(cat /proc/sys/crypto/fips_enabled)" = "1" ]; then
-        rpm -ivh --nodigest --nofiledigest $CONSUMER_RPM
+    if ! [ -x "$(command -v subscription-manager)" ] ; then
+      if [ "${VERSION_ID:0:1}" -gt 7 ]; then
```

Comment:
Why do you need to get a substring of the version here? It doesn't work no Fedora since `3 < 7` and will break once we get to RHEL 10.

Or is it to deal with `7.7`? If so, then I'd prefer to strip `.*` (note it's not a real regex, . is a literal value):
```suggestion
      if [ "${VERSION_ID%.*}" -gt 7 ]; then
```